src/app/ngrx/reducers/router.serializer.ts

```diff
@@ -0,0 +1,102 @@
+import { ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { RouterStateSerializer } from '@ngrx/router-store';
+
+import { IntervalModel, IntervalList, ChartType,
+  INTERVAL_HOURLY, METRICSTYPE_DEMOGRAPHICS, METRICSTYPE_DOWNLOADS, METRICSTYPE_TRAFFICSOURCES, MetricsType } from './models';
+
+import { getBeginEndDateFromStandardRange, getStandardRangeForBeginEndDate } from '../../shared/util/date/date.util';
```

Comment:
I am still trying to figure out what the circular dependency warning is from, but I've changed a bunch of imports around and am not getting it to clear.